File: 프로그래머스/2/42586. 기능개발/기능개발.cpp
#include <string>
#include <vector>
#include <deque>
#include <numeric>

using namespace std;

vector<int> solution(vector<int> progresses, vector<int> speeds) {
    int N = progresses.size();
    
    vector<int> answer;
    int start_idx = 0;
    while (accumulate(answer.begin(), answer.end(), 0) < N)
    {
        for (int i = start_idx; i < N; ++i)
        {
            progresses[i] += speeds[i];
        }
        
        int cnt = 0;
        for (int i = start_idx; i < N; ++i)
        {
            if (progresses[i] >= 100)
            {
                start_idx = i + 1;
                cnt++;
            }
            else
            {
                break;
            }
        }
        
        if (cnt > 0)
        {
            answer.push_back(cnt);
        }
    }

    
    return answer;
}

File: 프로그래머스/2/42586. 기능개발/기능개발.py
from collections import deque

def solution(progresses, speeds):
    progresses = deque(progresses); speeds = deque(speeds)
    
    result = []
    while progresses:
        
        for i in range(len(progresses)):
            progresses[i] += speeds[i]
        
        cnt = 0
        while progresses and progresses[0] >= 100:
            progresses.popleft(); speeds.popleft()
            cnt += 1
        if cnt != 0: result.append(cnt)
    return result